src/shared/services/addresses/addresses.service.ts

```diff
@@ -15,9 +16,40 @@ export class AddressesService {
 
   getAddressesForPostcode(postcode: string): Observable<Array<AddressModel>> {
     return this.http
-      .get(this.appConfig.getPostcodeLookupUrl().replace('${postcode}', postcode))
+      .get(this.appConfig.getPostcodeLookupUrl()
+        .replace('${postcode}', postcode))
       .pipe(
-        map(response => response.json())
-      );
+        map(res=> res.json().results)
+       ).pipe(
+        map(output => output.map(addresses=>
+            AddressesService.mapToAddressModel(addresses["DPA"]))
```

Comment:
what's this magic keyword `DPA` ? Perhaps better extract it to a constant with a more meaningful name?

Reply:
👍